### PR TITLE
[5.1] Memcached sessions: allow configuration of cache store to use

### PR DESCRIPTION
**Pull request by tomcastleman**
Just as with database and redis sessions the `connection` can be specified, for memcached sessions this PR allows the `store` for the session to be specified, rather than simply using the default `memcached` store.

Here is a suggested modification to `config/session.php`:

```
    /*
    |--------------------------------------------------------------------------
    | Session Cache Store
    |--------------------------------------------------------------------------
    |
    | When using the "memcached" session driver, you may specify a cache store
    | that should be used for these sessions. This should correspond to a
    | store in your cache configuration options which uses the memcached
    | driver.
    |
    */

    'memcached_store' => 'memcachedstorefoo',
```

### CODE REVIEW

File: src/Illuminate/Session/SessionManager.php [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 use Illuminate\Support\Manager;
 use Symfony\Component\HttpFoundation\Session\Storage\Handler\NullSessionHandler;
+use Illuminate\Cache\MemcachedStore;
```
Comment (GrahamCampbell) [inline src/Illuminate/Session/SessionManager.php]: Please sort these by length.

File: src/Illuminate/Session/SessionManager.php [hunk 2]
```diff
@@ -103,7 +105,18 @@ protected function createApcDriver()
 	 */
 	protected function createMemcachedDriver()
 	{
-		return $this->createCacheBased('memcached');
+		$store = $this->app['config']['session.memcached_store'];
+
+		// Verify store uses the memcached driver
+		$repository = $this->app['cache']->store($store);
+		if (!($repository->getStore() instanceof MemcachedStore))
```
Comment (GrahamCampbell) [inline src/Illuminate/Session/SessionManager.php]: Please fix cs.

Comment (tomcastleman) [inline src/Illuminate/Session/SessionManager.php]: Could you let me know what CS needs fixing here? Be useful to know for future too.

Looking at PSR-1 and the [Laravel standards](http://laravel.com/docs/master/contributions) I can't immediately see what's wrong.

Thanks

Comment (GrahamCampbell) [inline src/Illuminate/Session/SessionManager.php]: Look at laravel's existing code.

Comment (phanan) [inline src/Illuminate/Session/SessionManager.php]: I think Graham meant you don't need to wrap the instance check in brackets. Also, you need a space before and after `!`.

**Comment by tomcastleman** [comment]
@phanan thanks.

File: src/Illuminate/Session/SessionManager.php [hunk 3]
```diff
@@ -103,7 +105,18 @@ protected function createApcDriver()
 	 */
 	protected function createMemcachedDriver()
 	{
-		return $this->createCacheBased('memcached');
+		$store = $this->app['config']['session.memcached_store'];
+
+		// Verify store uses the memcached driver
```
Comment (GrahamCampbell) [inline src/Illuminate/Session/SessionManager.php]: Please use lower case here. It's not a sentence because it doesn't end with a `.`.

**Comment by tomcastleman** [comment]
How come this was closed?

**Comment by tomcastleman** [comment]
How come this was closed @taylorotwell ?

**Comment by tomcastleman** [comment]
Sorry @taylorotwell - could you advise why you closed this one please?

**Pull request closed without merging**